### ISSUE TITLE
Blazeface result is not correct on WASM backend

### ISSUE THREAD

**Issue by deepanshusharma012** [tensorflow/tfjs#5490]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js):
  Yes, I am using the estimateFace function of blazeface to detect the actual face in the image uploaded. Here is my code for reference:

```
  const detectFace = async () => {
    const p = document.getElementById('result');

    p.innerHTML = "Detecting ...";
    console.log('Loading model...')
    const model = await blazeface.load({scoreThreshold: 0.9});
    console.log('Loaded model...')

    var predictions;
    var t0, t1;

    for (j = 0; j < 5; j++) {
        var t0 = performance.now()
        // Pass in a video stream (or an image, canvas, or 3D tensor) to obtain an
        // array of detected faces from the MediaPipe graph. If passing in a video
        // stream, a single prediction per frame will be returned.
        predictions = await model.estimateFaces(canvas, false);
        console.log(predictions);
        var t1 = performance.now()
    }
    if (predictions.length > 0) {
        var prediction = predictions[0]
        var text = "The face detection completed in " + (t1 - t0) + " ms.<br>The face matched with predictions of " + predictions[0].probability[0];
        text += "<br>"
        text += `<strong>Right Eye:</strong> <br> x = ${prediction.landmarks[0][0]}; y = ${prediction.landmarks[0][1]}<br>`
        text += `<strong>Left Eye:</strong> <br> x = ${prediction.landmarks[1][0]}; y = ${prediction.landmarks[1][1]}<br>`
        text += `<strong>Nose:</strong> <br> x = ${prediction.landmarks[2][0]}; y = ${prediction.landmarks[2][1]}<br>`
        text += `<strong>Mouth:</strong> <br> x = ${prediction.landmarks[3][0]}; y = ${prediction.landmarks[3][1]}<br>`
        text += `<strong>Right Ear:</strong> <br> x = ${prediction.landmarks[4][0]}; y = ${prediction.landmarks[4][1]}<br>`
        text += `<strong>Left Ear:</strong> <br> x = ${prediction.landmarks[5][0]}; y = ${prediction.landmarks[5][1]}<br>`
        renderPrediction(predictions);
    } else {
        var text = "No face found in the image.";
    }
    p.innerHTML = text;
}
```

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
  This behaviour is reproducible across multiple OS of both desktop & mobile devices.

- TensorFlow.js installed from (npm or script link):
  https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@3.1.0

- TensorFlow.js version (use command below):
  3.1.0  

- Browser version:
  this behavior is reproducible across multiple Browser versions including latest Chrome 92

**Describe the current behavior**
I am using the estimateFace function of blazeface to detect the actual face in the image uploaded. It is giving `0.9804430603981018` probability when I placed the mobile device image to detect. If the face is present in the frame, then it detects properly the face and results are as expected. But if we don't have any face in the frame, in that case the blazeface detection results shows that face exists while it is Mobile device with `0.9804430603981018` probability with 0.9 scoreThreshold. (Attaching the image for reference)

**Describe the expected behavior**
It should show result of face detected only if any face is present in frame. If there is no face in frame, then there is no point to show face detected with such high probability i.e. of 98% for mobile device on frame.

**Standalone code to reproduce the issue**
You can use the following codepen link to reproduce: https://codepen.io/deepanshusharma012/pen/MWmRaLW
Attaching the mobile device pic on which the issue is reproducible, Also the blazeface models that I'm using for face detection.

`Issue Reproducible Snapshot:` 
![Screenshot from 2021-08-14 17-17-12](https://user-images.githubusercontent.com/41855523/129445669-3d1142af-9f78-49ad-b085-b82153f84f39.png)

`Image to reproduce the issue`
![Proctoring Snapshot - 2021-08-12T11_03_56 388Z](https://user-images.githubusercontent.com/41855523/129445676-30295d7a-04b3-4b12-9da5-3056877a7174.jpeg)

`Blazeface models used in face detection:`
[blazeface models.zip](https://github.com/tensorflow/tfjs/files/6986431/blazeface.models.zip)

**Comment by deepanshusharma012** [comment]
@lina128 @rthadur Hope you are doing great!
Is there any update on this issue or any ETA to resolve it? Actually this issue is a blocker for our application and detecting the false events.
It would be great if you respond on it and try to roll it's fix asap.

**Comment by lina128** [comment]
Hi @deepanshusharma012 , thank you for reporting this. I tested it with our own demo, same problem. Actually we're working on a new BlazeFace model, which is going to be more accurate. It will be released this quarter.

**Comment by deepanshusharma012** [comment]
The work is still going on this issue. I hope we'll be getting activity on this issue, so no need to mark it stalled.

**Comment by deepanshusharma012** [comment]
Hi @lina128, as you said the new BlazeFace model (more accurate one) will be released by Q3 quarter. So, are we still working on it, or it has released? or any ETA for the same.

**Comment by lina128** [comment]
Sorry, I actually meant Q4. I replied in September, and we are already thinking for Q4 :) Sorry for the confusion.

**Comment by deepanshusharma012** [comment]
@lina128 Is there any update on the new BlazeFace model?

**Comment by lina128** [comment]
Added @ahmedsabie who is working on it.

**Comment by deepanshusharma012** [comment]
@lina128 Thanks for the update. 

@ahmedsabie Do we have any ETA to get the new BlazeFace model (more accurate one) released?

**Comment by ahmedsabie** [comment]
hey @deepanshusharma012 we are aiming for sometime this month

**Comment by rthadur** [comment]
@deepanshusharma012 did you get chance to check latest blazeface model here https://github.com/tensorflow/tfjs-models/tree/master/pose-detection/src/blazepose_tfjs , it is now deployed.

**Comment by ayadav** [comment]
@rthadur will test shortly and share results.

**Comment by rthadur** [comment]
Closing this issue , please @mention to reopen, if the issue still persists or open a new issue with latest information. Thank you

**Comment by rahulgarg03** [comment]
Hi @rthadur , the above issue still persists for latest blazeface models having modelType set as 'short' (2 metres). Will there be any updates for the same? It is currently working fine for model type set as 'full'(5 metres).
Image for which the issue persists:
![Testing Image](https://user-images.githubusercontent.com/90387254/181225616-90c17216-9986-42b7-8720-4a15821578df.png)

**Comment by lina128** [comment]
We have deprecated blazeface API, please use face-detection instead: https://github.com/tensorflow/tfjs-models/tree/master/face-detection

**Comment by rahulgarg03** [comment]
Hi @lina128, I have used the face-detection model. But in this face-detection model for modelType set as 'short' (2 metres) the issue still persists. Will there be any updates for the same?

**Comment by lina128** [comment]
Thank you for testing out the new API, I will investigate.

**Comment by rahulgarg03** [comment]
Hi @lina128 Is there any update on the above issue?

**Comment by gaikwadrahul8** [comment]
Hi, @deepanshusharma012 

Apologize for the delayed response and we're re-visiting our older issues and checking whether those issues got resolved or not as of now so May I know are you still looking for the solution or your issue got resolved ?

If issue still persists after trying with latest version of TFJs and please refer this [official documentation](https://github.com/tensorflow/tfjs-models/tree/master/face-detection) please let us know with error log and code snippet to replicate the same issue from our end ?

Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

**Comment by SarthakBhatia2018** [comment]
Hi @gaikwadrahul8 , For our current use case we are majory targetting using this for webcam and front facing cameras on smartphone, so as per the documentation model type should be set as short. [https://drive.google.com/file/d/1d4-xJP9PVzOvMBDgIjz6NhvpnlG9_i0S/preview]

I have tested the same with the latest face detection models for above image. It is giving the same issues when tested for modelType as short. In case we set the model type as full it works correctly.

**Comment by SarthakBhatia2018** [comment]
Hi @gaikwadrahul8,

Checked the same with mediapipe-gpu, tfjs-webgl and tfjs-wasm backends. Attaching the results with images : 

**with mediapipe-gpu**
![2](https://github.com/tensorflow/tfjs/assets/48357816/89adcce2-5484-4ee8-9976-e682236abbde)

![1](https://github.com/tensorflow/tfjs/assets/48357816/9276ae6c-2b8a-4104-8098-c54b2444692f)


**with tfjs-wasm :-** 
![6](https://github.com/tensorflow/tfjs/assets/48357816/41d5ee24-78be-49ae-8348-039ab9c12528)

![5](https://github.com/tensorflow/tfjs/assets/48357816/4bfe73d0-a64a-4b1e-821d-21c4f1491d2c)

**with tfjs-webgl**
![4](https://github.com/tensorflow/tfjs/assets/48357816/84b4de92-7363-44c1-95e5-b9ab7ad69764)

![3](https://github.com/tensorflow/tfjs/assets/48357816/cf66a4e2-dc6c-4611-8e2f-a97efb6b8fc2)

**Comment by SarthakBhatia2018** [comment]
Hi @gaikwadrahul8 let me know if there is any update on above.

**Comment by gaikwadrahul8** [comment]
Hi, @SarthakBhatia2018 

Apologize for the delayed response and could you please help us with your codepen example which you tried with latest version of `@tensorflow/tfjs@4.8.0` or previous stable version of it and `@tensorflow-models/face-detection` so I'll try to replicate the same issue from my end ? Thank you!

**Comment by SarthakBhatia2018** [comment]
Hi @gaikwadrahul8,
I have added the code for testing face detection by uploading images. You can clone the following repository [https://github.com/SarthakBhatia2018/Face-Detection/]() and switch to upload_image brach. Navigate to face-detection -> demos -> upload_image. [https://github.com/SarthakBhatia2018/Face-Detection/tree/upload_image/face-detection/demos/upload_image]()

The steps to run it are the same as other folders in demos. Refer := [https://github.com/SarthakBhatia2018/Face-Detection/tree/upload_image/face-detection/demos#how-to-run-a-demo]()

**Comment by SarthakBhatia2018** [comment]
Hi @gaikwadrahul8 let me know if there is any update on the same.

**Comment by gaikwadrahul8** [comment]
Hi, @SarthakBhatia2018 

Apologize for the delayed response and I tried one random image from the internet with person face hiding behind the smartphone and in my case tfjs-webgpu runtime backend is giving me face detection as result so it seems like there some unexpected behaviour happening at the moment so we'll dig more into this issue and will update you soon , For your reference I have added screenshot below with `modelType :short`

Thank you for noticing this issue, I really appreciate your efforts and time. Thank you!

**Runtime backend : mediapipe-gpu** 

![image](https://github.com/tensorflow/tfjs/assets/115997457/bd934cf6-bbeb-4d64-ace9-7a690dcbe7cd)

**Runtime backend : tfjs-webgl**

![image](https://github.com/tensorflow/tfjs/assets/115997457/8e114452-f0a4-4098-afcc-f52cd423d3be)

**Runtime backend : tfjs-wasm**

![image](https://github.com/tensorflow/tfjs/assets/115997457/985a1d12-096f-4269-86fa-8ac3da409337)

**Runtime backend : tfjs-webgpu**

![image](https://github.com/tensorflow/tfjs/assets/115997457/4ea5fd53-b4ed-416f-aaed-bf490a0a395e)

**Comment by SarthakBhatia2018** [comment]
Hi @gaikwadrahul8, let me know if the above issue is fixed.

**Comment by nileshau-afk** [comment]
Hello @deepanshusharma012 ,

Thank you for bringing this to our attention and for providing a comprehensive overview. We appreciate the time you invested in creating this issue.

Currently, our internal development team is concentrated on other scheduled milestones, so we cannot prioritize this task immediately. However, we would love to see this move forward as a community-led effort.

If you’d like to submit a fix, please fork the repo and open a PR following the [TensorFlow Contribution Guidelines](https://www.tensorflow.org/community/contribute), [Tfjs Contributions](https://github.com/tensorflow/tfjs/blob/master/tfjs-vis/CONTRIBUTING.md#how-to-contribute) and be sure to link this issue in your PR to avoid duplicate work, kindly leave a comment here if you plan to take this on.

Thank you again for your interest and time.

**Comment by Varun-S10** [comment]
Hi @deepanshusharma012, Thank you for your patience. We have reproduced the issue.  As a workaround, you can switch from the default short-range model to the full-range model by setting `modelType` to `'full'` in your detector configuration options. The full-range model utilizes a deeper neural network architecture trained on a much wider variety of scenes, backgrounds, and objects, which completely eliminates these false detections while maintaining high accuracy for real human faces. Could you please try this and let us know if it works for you?
